### PR TITLE
Add documentation to more WML unit tests.

### PR DESCRIPTION
**Pull request by Pentarctagon**
---

Will merge after a couple days if there are no comments.

### CODE REVIEW

File: data/test/scenarios/filter_vision.cfg [hunk 1]
```diff
@@ -60,49 +60,7 @@
...
-            [/filter_location]
-        [/have_unit]
-    )}
-    [delay]
-        time=500
-    [/delay]
-#enddef
-
-#define assert_test_true X Y
+#define ASSERT_TEST_TRUE X Y
```
Comment (stevecotton) [inline data/test/scenarios/filter_vision.cfg:63]: `ASSERT_TEST_TRUE` sounds too much like a framework function. Maybe `ASSERT_CAN_SEE VIEWER TARGETS`?

File: data/test/scenarios/filter_vision.cfg [hunk 2]
```diff
@@ -148,22 +106,162 @@
...
-        {assert_test_true (side=3) (side=4)}
-        {assert_test_true (side=4) (side=1)}
-        {assert_test_true (side=4) (side=3)}
-        {assert_test_false (side=4) (side=2)}
-        {assert_test_false () (side=5)}
 
+        {ASSERT_TEST_FALSE () (side=5)}
         {SUCCEED}
     [/event]
 )}
```
Comment (stevecotton) [inline data/test/scenarios/filter_vision.cfg:267]: Please `undefine` macros that are only for use in this file.

File: data/test/scenarios/filter_vision.cfg (hunk 1 above)
Comment (Pentarctagon) [reply data/test/scenarios/filter_vision.cfg]: Updated.

File: data/test/scenarios/filter_vision.cfg (hunk 2 above)
Comment (Pentarctagon) [reply data/test/scenarios/filter_vision.cfg:267]: Updated.

**Pull request merged**